### PR TITLE
Allow users to seek by clicking on the transcript.

### PR DESCRIPTION
**Pull request by amuamushu**
With this change, when users click on the timestamp of a transcript line, the transcript jumps to that line. When the event handlers are added to the video and the discussion, they will also be seeked upon clicking the transcript timestamp.

### CODE REVIEW

File: src/main/webapp/scroll-container.js [hunk 1]
```diff
@@ -72,7 +72,8 @@ export class ScrollContainer extends HTMLDivElement {
    * Otherwise, does nothing.
    *
    */
-  scrollToTopOfContainer(element, forceScroll = false) {
+  scrollToTopOfContainer(
+      element = this.lastChild.firstChild, forceScroll = false) {
```
Comment (jaysaleh) [inline src/main/webapp/scroll-container.js:76]: What is element here?

File: src/main/webapp/view/transcript/transcript-seeker.js [hunk 2]
```diff
@@ -114,5 +117,7 @@ export default class TranscriptSeeker {
     return transcriptLinePointer;
   }
 
-  // TODO: Add a method that adds the eventListeners.
+  eventController() {
```
Comment (jaysaleh) [inline src/main/webapp/view/transcript/transcript-seeker.js:120]: Asking for context, what is this for?

File: src/main/webapp/scroll-container.js (hunk 1 above)
Comment (n-wach) [reply src/main/webapp/scroll-container.js:76]: the `element` within the container that should be scrolled to... maybe `childElement` would be better?  but it isn't a direct child either

File: src/main/webapp/view/transcript/transcript.js [hunk 3]
```diff
@@ -30,6 +30,9 @@ export function deleteTranscript() {
 export class TranscriptLineElement extends HTMLElement {
   static #DEFAULT_FONT_WEIGHT = 'text-muted';
   static #BOLD_FONT_WEIGHT = 'font-weight-bold';
+  static #TIME_OFFSET = 1;
+
+  #timestamp;
```
Comment (n-wach) [inline src/main/webapp/view/transcript/transcript.js:35]: can this be renamed `timestampElement`?  timestamp could be a lot of things

File: src/main/webapp/view/transcript/transcript.js [hunk 4]
```diff
@@ -54,9 +57,14 @@ export class TranscriptLineElement extends HTMLElement {
...
     const contentDivElement = TranscriptLineElement.createContentDivElement();
-    TranscriptLineElement.appendParagraphToContainer(
-        timestampRange, contentDivElement,
-        ['justify-content-start', 'mb-1', 'transcript-line-timestamp']);
+    this.#timestamp =
+        TranscriptLineElement.createParagraphWithClasses(timestampRange, [
+          'justify-content-start',
+          'mb-1',
+          'transcript-line-timestamp',
+          'timestamp',
```
Comment (n-wach) [inline src/main/webapp/view/transcript/transcript.js:65]: what's the difference between transcript-line-timestamp and timestamp?  Can these CSS classes be merged?

File: src/main/webapp/view/transcript/transcript.js [hunk 5]
```diff
@@ -94,14 +109,23 @@ export class TranscriptLineElement extends HTMLElement {
    * <p>Adds classes the the p tag if `classList` is provided.
    */
   static appendParagraphToContainer(text, container, classes = []) {
+    container.appendChild(
+        TranscriptLineElement.createParagraphWithClasses(text, classes));
+  }
+
+  /**
+   * Creates a p tag to store the given `text`.
```
Comment (n-wach) [inline src/main/webapp/view/transcript/transcript.js:117]: `p` tag

File: src/main/webapp/view/transcript/transcript.js [hunk 6]
```diff
@@ -94,14 +109,23 @@ export class TranscriptLineElement extends HTMLElement {
...
+   *
+   * <p>Adds classes the the p tag if `classList` is provided.
```
Comment (n-wach) [inline src/main/webapp/view/transcript/transcript.js:119]: I assume this should be `to the`? and add ticks on `p` please

File: src/main/webapp/view/transcript/transcript.js [hunk 7]
```diff
@@ -30,6 +30,9 @@ export function deleteTranscript() {
 export class TranscriptLineElement extends HTMLElement {
   static #DEFAULT_FONT_WEIGHT = 'text-muted';
   static #BOLD_FONT_WEIGHT = 'font-weight-bold';
+  static #TIME_OFFSET = 1;
```
Comment (n-wach) [inline src/main/webapp/view/transcript/transcript.js:33]: Can you add a comment explaining what this is for?  And include units if this is in seconds/milliseconds/etc.

File: src/main/webapp/scroll-container.js (hunk 1 above)
Comment (n-wach) [reply src/main/webapp/scroll-container.js:76]: Why was this change made?  I don't see scroll being called anywhere without an element specified

File: src/main/webapp/view/transcript/transcript.js (hunk 7 above)
Comment (amuamushu) [reply src/main/webapp/view/transcript/transcript.js]: done, thanks!

File: src/main/webapp/view/transcript/transcript-area.js [hunk 8]
```diff
@@ -71,15 +71,18 @@ export default class TranscriptArea {
...
     const transcriptContainer = TranscriptArea.transcriptScrollContainer();
     const ulElement = document.createElement('ul');
     // TODO: Move the class assignment to the HTML.
     ulElement.class = 'mx-auto';
     transcriptContainer.appendChild(ulElement);
     transcriptLines.forEach((transcriptLine) => {
-      ulElement.appendChild(
-          TranscriptLineElement.createTranscriptLineElement(transcriptLine));
+      const transcriptLineElement =
+          TranscriptLineElement.createTranscriptLineElement(transcriptLine);
```
Comment (billie-cat) [inline src/main/webapp/view/transcript/transcript-area.js:82]: Does this do any work?

File: src/main/webapp/view/transcript/transcript-seeker.js [hunk 9]
```diff
@@ -114,5 +117,10 @@ export default class TranscriptSeeker {
     return transcriptLinePointer;
   }
 
-  // TODO: Add a method that adds the eventListeners.
+  /**
+   * Returns the eventController.
+   */
```
Comment (billie-cat) [inline src/main/webapp/view/transcript/transcript-seeker.js:122]: This comment isn't useful. I would expand on what the event controller does.

File: src/main/webapp/view/transcript/transcript-area.js (hunk 8 above)
Comment (amuamushu) [reply src/main/webapp/view/transcript/transcript-area.js:82]: It calls the constructor for creating a `transcriptLineElement` [link to code](https://github.com/googleinterns/zoomtube/blob/master/src/main/webapp/view/transcript/transcript.js#L46)

File: src/main/webapp/view/transcript/transcript-seeker.js (hunk 9 above)
Comment (amuamushu) [reply src/main/webapp/view/transcript/transcript-seeker.js:123]: gotcha, updated!

File: src/main/webapp/view/transcript/transcript-area.js (hunk 8 above)
Comment (billie-cat) [reply src/main/webapp/view/transcript/transcript-area.js:101]: For some reason I thought this was part of the constructor.

**Pull request merged**